### ISSUE TITLE
Commenting on Project Code

### ISSUE THREAD

**Issue by JasonCP14** [nus-cs2030/2122-s1#306]
Hi, I was wondering if our project’s coding style is also graded, because it didn’t pass the checkstyle test on my ubuntu due to not having javadoc comments

**Comment by Elyxpenguin** [comment]
Yes coding style is graded, the last testcase on codecrunch checks for style but I don't think javadoc comments are required as I got A on codecrunch without writing javadoc comments.

**Comment by albertsutz** [comment]
The checkstyles will require javadoc for public methods. So I guess you would need to do those. I don't think that the other methods should need javadoc as non-public methods are not visible to client anyway?

**Comment by CalebLee** [comment]
I think public classes should have javadocs as well, although CodeCrunch doesn't check for them at the moment. Does anyone know if Main is considered a public class? And is there really a need to write javadocs for toString since it's public o.O

**Comment by JasonCP14** [comment]
Actually I passed all test cases already without writing javadoc comments. What i’m really asking is whether there is a difference in the final mark with and without it

**Comment by kxshxsh** [comment]
my TA said that no need for javadocs..

**Comment by teominghuang** [comment]
My TA said that javadocs is required for public classes, constructors and methods

**Comment by upabove122** [comment]
Im sure the prof mentioned in a lecture that no java doc is required

**Comment by throwawayAcc123** [comment]
How do you guys know that test case #32 is testing for style?

**Comment by kxshxsh** [comment]
> How do you guys know that test case #32 is testing for style?

It points out checkstyle errors in our program files

**Comment by kxshxsh** [comment]
> My TA said that javadocs is required for public classes, constructors and methods

Ask him to recheck with prof, mine initially said yes for javadocs too but then when I told him about the discussions here he said he recalled prof saying that he disabled the requirements for java docs

**Comment by jehousoh** [comment]
as long as it gets an A on CodeCrunch then it's fine

**Comment by urcs2030** [comment]
thanks for the comments
